runtime/test/particle-api-test.js

```diff
@@ -148,7 +148,7 @@ describe('particle-api', function() {
     await util.assertSingletonIs(newView, Result, 'success');
   });
 
-  it('can load a recipe', async () => {
+  const testInnerRecipe = async innerRecipe => {
```

Comment:
So there's: https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

One takeaway I get from that is that tests should concretely present input and output pairs, and that utilities should help emphasize these rather than obscuring them.

I'm a bit uncomfortable about this particular utility because, while it makes the input clear it doesn't show what the output is supposed to be in the places where it's used.

The original 'can load a recipe' test was ensuring that setting inView resulted in outView being set, which is mainly a consequence of the fact that we were using PassThrough, which is already not great in terms of matching inputs to outputs. Moving the mechanics of that testing away from the body of the recipe makes it much harder to see what's going on.

For now I think it's best just to duplicate the testing logic, to keep it adjacent to the recipes. As a concrete long-term goal, I think we need to develop a way to treat a running arc as a simple thing with APIs that let us 'peek' and 'poke' values. That will make these tests more concise.

Reply:
Yeah, this was a bit of a hack test factoring primarily aimed at just not copying so many LOC. Agree that this obscures more than is perhaps wise. I will revise to just dupe for now before merge.